src/solvers/dgsem_p4est/containers_2d_manifold_in_3d.jl

```diff
@@ -261,4 +272,142 @@ function calc_contravariant_vectors_2d_shell!(contravariant_vectors::AbstractArr
 
     return contravariant_vectors
 end
+
+# Calculate contravariant vectors, multiplied by the Jacobian determinant J of the transformation mapping,
+# using the invariant curl form for a cubed sphere.
+# In the cubed sphere, the node coordinates are first mapped with polynomials in ξ and η, and then we add 
+# a linear ζ dependency, i.e.:
+#    xₗ(ξ, η, ζ) = ζ * xₗ(ξ, η) with ζ = 1 at the sphere surface.
+# 
+# As a result, the covariant vectors with respect to ζ are xₗ_ζ = xₗ
+#
+# We compute the metric terms in two steps
+# 1. Compute the 3D contravariant vectors, Jaⁱ=J*grad(ξⁱ), using the curl invariant form and xₗ_ζ = xₗ
+# 2. Convert the 3D mapping Jacobian determinant J:=a_k.(a_i x a_j) to 2D J_2D=||a_i x a_j||
+function calc_contravariant_vectors_2d_shell!(contravariant_vectors::AbstractArray{<:Any,
+                                                                                   5},
+                                              element,
+                                              jacobian_matrix, node_coordinates,
+                                              basis::LobattoLegendreBasis,
+                                              metric_terms::MetricTermsCurlInvariant)
+    @unpack derivative_matrix = basis
+
+    # 1. Compute the 3D contravariant vectors, Jaⁱₙ=J*grad(ξ), using the curl invariant form and xₗ_ζ = xₗ
+    # The general form is
+    # Jaⁱₙ = 0.5 * ( ∇ × (Xₘ ∇ Xₗ - Xₗ ∇ Xₘ) )ᵢ  where (n, m, l) cyclic and ∇ = (∂/∂ξ, ∂/∂η, ∂/∂ζ)ᵀ
+
+    for n in 1:3
+        # (n, m, l) cyclic
+        m = (n % 3) + 1
+        l = ((n + 1) % 3) + 1
+
+        # Calculate Ja¹ₙ = 0.5 * [ (Xₘ Xₗ_ζ - Xₗ Xₘ_ζ)_η - (Xₘ Xₗ_η - Xₗ Xₘ_η)_ζ ]
+        # For each of these, the first and second summand are computed in separate loops
+        # for performance reasons.
+
+        # First summand 0.5 * (Xₘ Xₗ_ζ - Xₗ Xₘ_ζ)_η
+        @turbo for j in eachnode(basis), i in eachnode(basis)
```

Comment:
The LoopVectorization.jl `@turbo` macro is used here to compute the metrics, but it is not used anywhere else (e.g. in the volume integral). With the [future support for LoopVectorization.jl uncertain](https://discourse.julialang.org/t/why-is-loopvectorization-deprecated/109547), I am hesitant to use it in new code unless there's a very significant speedup.

Reply:
I adopted this from the main Trixi code when I adapted the invariant curl metrics to work on a 2D manifold. Since the use of `@turbo` is expected to improve the performance of the (re)computation of the metric terms when AMR is activated (not yet included in this PR), I suggest we keep it, as in Trixi. What do you think?

Reply:
I tried comparing to a version without `@turbo` here to see if it would affect the performance. It turns out that `@turbo` actually improves performance for me a lot more than I thought it would (I assumed the compiler would automatically do something similar for this type of loop), so perhaps it should be kept in. But in my opinion, we if we add use `@turbo` here, we should also consider using it in the volume integrals, which are at least as performance critical. 

When I asked about this in the Slack channel, @sloede said that local optimizations were left out of the main kernels in Trixi.jl for the moment, in order to avoid making the core algorithms more complicated than necessary. I am not sure if this argument also applies to a more specialized package such as TrixiAtmo.jl, though.